cli/npm/installer/common/mod.rs

```diff
@@ -27,48 +29,56 @@ pub trait NpmPackageFsInstaller: std::fmt::Debug + Send + Sync {
   ) -> Result<(), JsErrorBox>;
 }
 
-pub trait NpmPackageExtraInfoProvider: std::fmt::Debug + Send + Sync {
-  async fn get_package_extra_info(
-    &self,
-    package_id: &PackageNv,
-    package_path: &Path,
-    expected: ExpectedExtraInfo,
-  ) -> Result<deno_npm::NpmPackageExtraInfo, JsErrorBox>;
+pub struct CachedNpmPackageExtraInfoProvider {
+  inner: Arc<NpmPackageExtraInfoProvider>,
+  cache: RwLock<rustc_hash::FxHashMap<PackageNv, NpmPackageExtraInfo>>,
 }
```

Comment:
Separated the caching to separate the concerns (and it allows for a long lived service and short lived cached)